php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -68,6 +68,10 @@ public function __construct($desc = NULL)
         // specific descriptor from the descriptor pool.
         if (get_class($this) === 'Google\Protobuf\Internal\MapEntry') {
             $this->desc = $desc;
+            foreach ($desc->getField() as $field) {
+                $setter = $field->getSetter();
+                $this->$setter($this->defaultValue($field));
```

Comment:
Thanks for the catch. Do you know how to trigger the Strict Standards error?
For normal messages, a primitive field's setter is passing-by-value while a message field's setter is passing-by-message.
MapEntry is a special message for internal implementation only. Its interface need to be general enough for whether primitive field or message field. However, I think you are right. Change the setter parameter to pass-by-value seems better to me.